### ISSUE TITLE
Add dedicated panel for managing failed imports

### ISSUE THREAD

**Issue by AndyBarcia** [godotengine/godot-proposals#2481]
### Describe the project you are working on

The godot editor.

### Describe the problem or limitation you are having in your project

Managing failed imports is slightly annoying. As described in [this closed proposal](https://github.com/godotengine/godot/issues/34945) that was closed in the main tracker: the editor will try to reimport broken files whenever it regains focus, which fills the output with an overwhelming amount of error messages, making it harder to understand.

Also, when there are multiple failed imports, it becomes a bit tricky to understand what's going on just from the output of console, and it would be nice to have a dedicated panel to manage failed imports (specially for beginners). 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

One thing that could be done is that when the editor first opens, if it encounters errors it could show something like this:

![example3](https://user-images.githubusercontent.com/40731413/111867354-afe1fe80-8973-11eb-9023-1dcbf20395ce.png)

Which I think that it's far better for beginners than what we have now:

![ds](https://user-images.githubusercontent.com/40731413/111867727-2d0e7300-8976-11eb-9103-04b85d90213a.png)

The panel would enable to open each individual file in the file manager, open its import dock, and maybe a couple more things (like "Move to Trash", or "Reimport"). In addition, this panel could also show the specific error that each import encountered. But I think that this would be trickier to do, as the way that errors are reported would need to change slightly.

It would also be useful if this panel could be manually opened. But I'm not sure where that "Open Failed Import Manager" button should go (maybe as an editor tool??).

I also designed this other mockup that shows failed imports individually, but I don't like it as much:

![example2](https://user-images.githubusercontent.com/40731413/111867532-f126de00-8974-11eb-8033-b5ebfb1ac638.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I'm still not sure of the exact implementation. I haven't looked at any code yet. But I could try to implement it if nobody opposes :)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Probably not.

### Is there a reason why this should be core and not an add-on in the asset library?

I think this is an interesting feature to have on core.

**Comment by AndyBarcia** [comment]
My current design 
![example](https://user-images.githubusercontent.com/40731413/112539112-ddcea500-8db0-11eb-9be7-6714278decf2.png)

**Comment by Ev1lbl0w** [comment]
This is especially interesting for showing what exact errors are happening. I just remembered https://github.com/godotengine/godot/issues/46486 when looking at this, and I believe this would be the perfect way to warn the user about this problem.

**Comment by AndyBarcia** [comment]
> showing what exact errors are happening

What do you mean with this, exactly? I was thinking that the right panel could show a brief generic description of each error type. Are you suggesting to include more specific information based on the resource type too? So, for example, when an ERR_FILE_CORRUPT error is encountered with an audio file it also warns the user to check if the file extension is correct? 

Also, not related to your comment. But I was wondering if it would be a good idea to automatically show this panel when a project with import errors is opened (just once at the start of the project). Or would that be too annoying?

**Comment by Ev1lbl0w** [comment]
> Are you suggesting to include more specific information based on the resource type too?

Yes, I believe this UI would also be a good tool to better explain to the user what exactly failed during the importing process. For example, the description for `ERR_FILE_CORRUPT` could show a generic description, and then a specific message for the file *(for example, "This file has .ogg (Ogg Vorbis) extension, but it's actually a MP3 audio file", something like that)*

> Also, not related to your comment. But I was wondering if it would be a good idea to automatically show this panel when a project with import errors is opened (just once at the start of the project). Or would that be too annoying?

Maybe you could show a popup at the end of the import process, saying something like "There are 39 files that failed to import. Do you wish to open the Failed Imports Manager to manage them?", having two buttons to either manage the broken files or simply ignore them and close the popup.
